Add new option `reevaluate`

vtuan10 opened the pull request:
Thank you very much for this awesome plugin. It improved our CI pipeline by far. 

This PR adds the following changes:
- Remove blacklist projects before evaluating the dependents: In our specific case, we have a module defined which only has lint rules. As this module is not a `com.android.library` or `com.android.application` it misses tasks like unit tests. This results in warnings, even when this module is added to the blacklist 
- Exclude test configuration when evaluating project dependents: This fixes #3 as you could run into cyclic dependencies because the configuration is not considered. Example: 
`databaseModule`: `testImplementation(project(:sharedTestModule))`
`sharedTestModule`: `implementation(project(:databaseModule))`
This is possible, as the `databaseModule` has only a dependency in its tests, while the `sharedTestModule` needs database functionality for testing purposes. 
- Reevaluate option: As the test depedencies are not considered for evaluation, there has to be an option to trigger all tasks in case the `sharedTestModule` has changes. 

I was unable to test these changes locally, as I am not versed enough in gradle to get this plugin to run for development purposes in our project. Hopefully, @ismaeldivita can confirm the changes work.

ismaeldivita commented:
Thanks for the PR. I'll try to take a look today

ismaeldivita commented:
I can't see how removing the blacklist before evaluating the dependents will help with the logs. Still see them.
And actually this will create some bugs. 

For example: 
A <-- B <-- C

If `C` is blacklisted and contains changes you still expect the tasks run on `A` and `B`. Removing the evaluation of dependents on blacklisted modules will prevent this.

About the logs, I still see some value of printing missing tasks for a module, this could help highlight/debug a problem. But I agree it's really annoying for most cases, so I'm thinking to put this into a different log level for the next release.

All the rest looks good to me, just revert the `dfd7fef`.

Another thing, maybe worth put something on the README talking about the test cyclic dependency and the `reevaluate` usage for these cases. If you want to create a "Notes" section before the LICENSE feel free. Just make sure to put under `<!---` / `--->` while we waiting for the release.

Again, thanks for the PR

vtuan10 commented:
Thanks, I removed commit `dfd7fef`. 
We currently work around this by defining empty tasks in the mentioned module 😆 

I also added a little description about cyclic test dependency to the `README.md`.
I'm not sure if I understood you correctly, but I commented in the `README.md` all changes which would come with this PR (reevaluate and the description of Cyclic Test Dependency) out. Feel free to change/rewrite/delete my changes to the README.md 😄
